src/app/components/category-products/category-products.component.ts

```diff
@@ -202,5 +208,29 @@ export class CategoryProductsComponent {
     )
     .subscribe();
 
+    onCheckChange(event:Event) {
+      const formArray: FormArray = this.searchForm.get('ratingsArray') as FormArray;
+      const eventTarget = event.target as HTMLInputElement
 
+      /* Selected */
+      if(eventTarget.checked){
+        // Add a new control in the arrayForm
+        formArray.push(new FormControl(eventTarget.value));
+      }
+      /* unselected */
+      else{
+        // find the unselected element
+        let i: number = 0;
+    
+        formArray.controls.forEach((ctrl: AbstractControl) => {
```

Comment:
tutaj w przykładzie było żeby ctrl oznaczyc jako FormControl, ale nie działało i musiałam użyć AbstractControl

https://stackoverflow.com/questions/40927167/angular-reactiveforms-producing-an-array-of-checkbox-values